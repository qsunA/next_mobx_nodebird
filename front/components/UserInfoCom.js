import React from 'react';
import { Card, Avatar } from 'antd';

const UserInfoCom = ({userInfo})=>{
    return(
        <Card
                    actions={[
                        <div key="twit">짹짹<br/>{userInfo.Posts}</div>,
                        <div key="following">팔로잉<br/>{userInfo.Followings}</div>,
                        <div key="follower">짹짹<br/>{userInfo.Followers}</div>,
                    ]}
                >
                    <Card.Meta
                        avatar={<Avatar>{userInfo.nickname[0]}</Avatar>}
                        title={userInfo.nickname}
                    />
                </Card>
    );
}

export default UserInfoCom;