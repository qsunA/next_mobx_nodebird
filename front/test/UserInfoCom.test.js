import React from 'react';
import { mount } from 'enzyme';
import UserInfoCom from '../components/UserInfoCom';

const userInfo = {
    Posts : 10, 
    Followings : 2,
    Followers : 5,
    nickname : 'test'
};

describe('<UserInfoCom/>', ()=>{
    it('matches snapshot',()=>{
        const wrapper = mount(<UserInfoCom userInfo={userInfo}/>);
        expect(wrapper).toMatchSnapshot();
    });
    it('renders userInfo',()=>{
        const wrapper = mount(<UserInfoCom userInfo={userInfo}/>);
        expect(wrapper.props().userInfo.Posts).toBe(10);
        expect(wrapper.props().userInfo.nickname).toBe('test');
        const avatar= wrapper.find('Avatar');
        expect(avatar.contains('t')).toBe(true);
    });
})