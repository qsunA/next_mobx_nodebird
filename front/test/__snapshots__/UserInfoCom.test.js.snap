// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserInfoCom/> matches snapshot 1`] = `
<UserInfoCom
  userInfo={
    Object {
      "Followers": 5,
      "Followings": 2,
      "Posts": 10,
      "nickname": "test",
    }
  }
>
  <Card
    actions={
      Array [
        <div>
          짹짹
          <br />
          10
        </div>,
        <div>
          팔로잉
          <br />
          2
        </div>,
        <div>
          짹짹
          <br />
          5
        </div>,
      ]
    }
  >
    <div
      className="ant-card ant-card-bordered"
    >
      <div
        className="ant-card-body"
        style={Object {}}
      >
        <Meta
          avatar={
            <Avatar
              shape="circle"
              size="default"
            >
              t
            </Avatar>
          }
          title="test"
        >
          <div
            className="ant-card-meta"
          >
            <div
              className="ant-card-meta-avatar"
            >
              <Avatar
                shape="circle"
                size="default"
              >
                <span
                  className="ant-avatar ant-avatar-circle"
                  style={Object {}}
                >
                  <span
                    className="ant-avatar-string"
                  >
                    t
                  </span>
                </span>
              </Avatar>
            </div>
            <div
              className="ant-card-meta-detail"
            >
              <div
                className="ant-card-meta-title"
              >
                test
              </div>
            </div>
          </div>
        </Meta>
      </div>
      <ul
        className="ant-card-actions"
      >
        <li
          key="action-0"
          style={
            Object {
              "width": "33.333333333333336%",
            }
          }
        >
          <span>
            <div
              key="twit"
            >
              짹짹
              <br />
              10
            </div>
          </span>
        </li>
        <li
          key="action-1"
          style={
            Object {
              "width": "33.333333333333336%",
            }
          }
        >
          <span>
            <div
              key="following"
            >
              팔로잉
              <br />
              2
            </div>
          </span>
        </li>
        <li
          key="action-2"
          style={
            Object {
              "width": "33.333333333333336%",
            }
          }
        >
          <span>
            <div
              key="follower"
            >
              짹짹
              <br />
              5
            </div>
          </span>
        </li>
      </ul>
    </div>
  </Card>
</UserInfoCom>
`;
